### PR TITLE
doc: Updated doc/dev/README.md with relative links

### PR DESCRIPTION
**Pull request by mjonss**
For easier navigation in github.

### What problem does this PR solve?
Fix the links in the doc/dev/README.md file, to use relative links that work on github.

### What is changed and how it works?
Only adding '../../' for links and correcting some.

Tests
 - Manual test (add detailed scripts or steps below)
 Checked the links on github
 - No code

Side effects
None (no code change)

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: doc/dev/README.md [hunk 1]
```diff
@@ -61,4 +61,4 @@ Some key concepts:
 
 ### Manifest formats
 
-See `ComponentManifest` and `VersionManifest` data structures in [component.go](pkg/repository/component.go) and [version.go](pkg/repository/version.go).
+See `ComponentManifest` and `VersionManifest` data structures in [component.go](../../server/handler/component.go) and [version.go](../../pkg/version/version.go).
```
Comment (AstroProfundis) [inline doc/dev/README.md:64]: This line seems out-dated, the manifest structures are defined in `pkg/repository/types.go`.

Comment (lucklove) [inline doc/dev/README.md:64]:
```suggestion
See `ComponentManifest` and `VersionManifest` data structures in [component.go](../../pkg/repository/types.go) and [version.go](../../pkg/version/version.go).
```

**Pull request merged**